website/docs/indexing.md

```diff
@@ -20,34 +24,90 @@ _Figure: Comparison of merge cost for updates (yellow blocks) against base files
+
+#### Flink based configs
+
+For Flink DataStream and Flink SQL only support Bucket Index and internal Flink state store backed in memory index. 
+Following are the basic configs that control the indexing behavior. Please refer [here](https://hudi.apache.org/docs/next/configurations#Flink-Options-advanced-configs)
+for advanced configs.
+
+| Config Name                                                                       | Default                                                                                         | Description                                                                                                                                                                                                                                                                            |
+| ----------------------------------------------------------------------------------| ----------------------------------------------------------------------------------------------- |----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| index.type                                                                        | FLINK_STATE (Optional)                  | Index type of Flink write job, default is using state backed index. Possible values:<br /> <ul><li>FLINK_STATE</li><li>BUCKET</li></ul><br />  `Config Param: INDEX_TYPE`                                                                                                              |
+| hoodie.index.bucket.engine                                                        | SIMPLE (Optional)                                                                               | org.apache.hudi.index.HoodieIndex$BucketIndexEngineType: Determines the type of bucketing or hashing to use when `hoodie.index.type` is set to `BUCKET`.    Possible Values: <br /> <ul><li>SIMPLE</li><li>CONSISTENT_HASHING</li></ul>|
+
+
+
+
+## Indexing Strategies
+
 Since data comes in at different volumes, velocity and has different access patterns, different indices could be used for different workload types.
-Let’s walk through some typical workload types and see how to leverage the right Hudi index for such use-cases. 
+Let’s walk through some typical workload types and see how to leverage the right Hudi index for such use-cases.
 This is based on our experience and you should diligently decide if the same strategies are best for your workloads.
 
-## Indexing Strategies
```

Comment:
I do not want to expand the scope here. Will let you take the call. But we should add streaming writes workload type and call out that RLI will be the best. 
for eg, if table size is 1TB, but incremental ingestion brings in 1% of data (1Gb or less), RLI will give the best performance out of any other global index options. 
Also, for update heavy workloads in case of global index, RLI will out perform other indexes.

Reply:
@nsivabalan  Agree. We should consider a second doc PR change for this part. Since this involves some more substantiation. I can collab with you next week to add this.

Reply:
sure